File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/data/mapper/PostMapper.java
package com.example.postscommentsapp.data.mapper;

import com.example.postscommentsapp.data.local.entity.PostEntity;
import com.example.postscommentsapp.data.remote.dto.PostDto;
import com.example.postscommentsapp.domain.model.Post;

import java.util.ArrayList;
import java.util.List;

public class PostMapper {

    public static PostEntity toEntity(PostDto dto) {
        return new PostEntity(
                dto.getId(),
                dto.getTitle(),
                dto.getBody()
        );
    }

    public static Post toDomain(PostEntity entity) {
        return new Post(
                entity.getId(),
                entity.getTitle(),
                entity.getBody()
        );
    }

    public static List<PostEntity> toEntityList(List<PostDto> list) {
        List<PostEntity> entities = new ArrayList<>();
        for (PostDto dto : list) {
            entities.add(toEntity(dto));
        }
        return entities;
    }

    public static List<Post> toDomainList(List<PostEntity> list) {
        List<Post> result = new ArrayList<>();
        for (PostEntity entity : list) {
            result.add(toDomain(entity));
        }
        return result;
    }
}

File: PostsAndComments/app/src/main/java/com/example/postscommentsapp/data/mapper/PostMapper.kt
package com.example.postscommentsapp.data.mapper

import com.example.postscommentsapp.data.local.entity.PostEntity
import com.example.postscommentsapp.data.remote.dto.PostDto
import com.example.postscommentsapp.domain.model.Post

object PostMapper {

    fun fromDto(dto: PostDto): Post {
        return Post(
            id = dto.id,
            title = dto.title,
            body = dto.body
        )
    }

    fun fromDtoList(list: List<PostDto>): List<Post> =
        list.map { fromDto(it) }

    fun toEntity(post: Post): PostEntity {
        return PostEntity(
            post.id,
            post.title,
            post.body
        )
    }

    fun toEntityList(list: List<Post>): List<PostEntity> =
        list.map { toEntity(it) }

    fun toDomain(entity: PostEntity): Post {
        return Post(
            id = entity.id,
            title = entity.title,
            body = entity.body
        )
    }

    fun toDomainList(list: List<PostEntity>): List<Post> =
        list.map { toDomain(it) }
}